psi4/src/psi4/adc/adc.cc

```diff
@@ -45,53 +45,48 @@ ADCWfn::ADCWfn(SharedWavefunction ref_wfn, Options& options) : Wavefunction(opti
     module_ = "adc";
 
     std::vector<std::string> irreps_ = molecule_->irrep_labels();
-    aoccpi_ = new int[nirrep_];
-    boccpi_ = new int[nirrep_];
-    avirpi_ = new int[nirrep_];
-    bvirpi_ = new int[nirrep_];
+    aoccpi_ = nalphapi_ - frzcpi_;
+    boccpi_ = nbetapi_ - frzcpi_;
+    avirpi_ = nmopi_ - frzvpi_ - nalphapi_;
+    bvirpi_ = nmopi_ - frzvpi_ - nbetapi_;
+
+    int naocc = aoccpi_.sum();
+    int nbocc = boccpi_.sum();
+    int navir = avirpi_.sum();
+    int nbvir = bvirpi_.sum();
 
-    int naocc = 0, nbocc = 0, navir = 0, nbvir = 0;
     int aoccount = 0, boccount = 0, avircount = 0, bvircount = 0;
-    for (int h = 0; h < nirrep_; h++) {
-        aoccpi_[h] = doccpi_[h] + soccpi_[h] - frzcpi_[h];
-        boccpi_[h] = doccpi_[h] - frzcpi_[h];
-        avirpi_[h] = nmopi_[h] - doccpi_[h] - soccpi_[h] - frzvpi_[h];
-        bvirpi_[h] = nmopi_[h] - doccpi_[h] - frzvpi_[h];
-
-        naocc += aoccpi_[h];
-        nbocc += boccpi_[h];
-        navir += avirpi_[h];
-        nbvir += bvirpi_[h];
-    }
+
     aocce_ = new double[naocc];
     bocce_ = new double[nbocc];
     avire_ = new double[navir];
     bvire_ = new double[nbvir];
 
-    nopen_ = 0;
-    for (int h = 0; h < nirrep_; h++) nopen_ += soccpi_[h];
+    nopen_ = soccpi().sum();
     if (!psio_) {
         throw PSIEXCEPTION("The wavefunction passed in lacks a PSIO object, crashing ADC. See GitHub issue #1851.");
     }
     if (nopen_) throw PSIEXCEPTION("Openshell calculation has not been implemented yet!");
```

Comment:
```suggestion
     if (nopen_) throw PSIEXCEPTION("Open-shell calculation has not been implemented yet!");
```

Reply:
Once again, I'll bother with this if I need to make other commits, but not otherwise. As ADC is not going to survive 1.7, this change isn't worth going through another test cycle.